getndazn/kopytko-unit-testing-framework
# feat: add mockFunction and other testFunctions

BREAKING CHANGE: adds insertKopytkoUnitTestSuiteArgument bs_const flag


## What did you implement:

Implemented:
- test function can now return an array of asserts
- mockFunction - operates on the test suite `__mocks` object
- it - shorthand for ts.addTest
- itEach - shorthand for ts.addParameterizedTests
- beforeAll - shorthand for ts.setBeforeAll
- beforeEach - shorthand for ts.setBeforeEach
- afterEach - shorthand for ts.setAfterEach
- afterAll - shorthand for ts.setAfterAll

Added documentation



## How did you implement it:

Added new functions.



## How can we verify it:

Run unit tests.
There is one test suite where all test cases are supposed to fail.

Tested on Roku DAZN app, to verify if changes break anything - everything seems to work.



## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** YES
